src/gui/guiFormSpecMenu.cpp

```diff
@@ -1624,23 +1624,57 @@ void GUIFormSpecMenu::parseListColors(parserData* data, const std::string &eleme
 void GUIFormSpecMenu::parseTooltip(parserData* data, const std::string &element)
 {
 	std::vector<std::string> parts = split(element,';');
-	if (parts.size() == 2) {
-		std::string name = parts[0];
-		m_tooltips[name] = TooltipSpec(utf8_to_wide(unescape_string(parts[1])),
-			m_default_tooltip_bgcolor, m_default_tooltip_color);
+	if (parts.size() < 2) {
+		errorstream << "Invalid tooltip element(" << parts.size() << "): '"
+				<< element << "'"  << std::endl;
 		return;
 	}
 
-	if (parts.size() == 4) {
-		std::string name = parts[0];
-		video::SColor tmp_color1, tmp_color2;
-		if ( parseColorString(parts[2], tmp_color1, false) && parseColorString(parts[3], tmp_color2, false) ) {
-			m_tooltips[name] = TooltipSpec(utf8_to_wide(unescape_string(parts[1])),
-				tmp_color1, tmp_color2);
-			return;
-		}
+	// Get mode and check size
+	bool rect_mode = parts[0].find(",") != std::string::npos;
+	size_t base_size = rect_mode ? 3 : 2;
+	if (parts.size() != base_size && parts.size() != base_size + 2) {
+		errorstream << "Invalid tooltip element(" << parts.size() << "): '"
+				<< element << "'"  << std::endl;
+		return;
+	}
+
+	// Read colors
+	video::SColor tmp_color1 = m_default_tooltip_bgcolor;
+	video::SColor tmp_color2 = m_default_tooltip_color;
+	if (parts.size() == base_size + 2 &&
+			(!parseColorString(parts[base_size], tmp_color1, false) ||
+				!parseColorString(parts[base_size + 1], tmp_color2, false))) {
+		errorstream << "Invalid color in tooltip element(" << parts.size()
+				<< "): '" << element << "'"  << std::endl;
+		return;
+	}
+
+	// Make tooltip spec
+	std::string text = unescape_string(parts[rect_mode ? 2 : 1]);
+	TooltipSpec spec(utf8_to_wide(text), tmp_color1, tmp_color2);
+
+	// Add tooltip
+	if (rect_mode) {
+		std::vector<std::string> v_pos = split(parts[0],',');
+		std::vector<std::string> v_geom = split(parts[1],',');
+
+		MY_CHECKPOS("tooltip",  0);
+		MY_CHECKGEOM("tooltip", 1);
+
+		v2s32 pos = padding + AbsoluteRect.UpperLeftCorner + pos_offset * spacing;
```

Comment:
Floating point values might be wanted.

Reply:
No, this is how all elements are positioned. This is pixel space, not inventory space

Reply:
Good to know, thanks.

Reply:
I'd like to test this though, as I've made this mistake before